Objective-C/CBLScope.h

```diff
@@ -0,0 +1,55 @@
+//
+//  CBLScope.h
+//  CouchbaseLite
+//
+//  Copyright (c) 2022 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+@class CBLCollection;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol CBLScope <NSObject>
```

Comment:
I think it's unnecessary for iOS to have protocol for CBLScope and Could be CBLCollection as well. 

Even though the spec said "Protocol", it's up to the platform to decide whether to have it or not. I believe Java will not have it as protocol. Let evaluate this when you implement it.